R/sheets_speedread.R

```diff
@@ -0,0 +1,221 @@
+#' Read Sheet as CSV
+#'
+#' @description
+#' This function uses a quick-and-dirty method to read a Sheet that bypasses the
+#' Sheets API entirely. It can be much faster than [sheets_read()] -- noticeably
+#' so for "large" spreadsheets. `sheets_speedread()` forms a special URL that
+#' reads Sheet values via CSV, then sends that data through vroom or readr.
+#' Here are the limitations we must accept to get faster reading:
+#' * Only cell values are available, not formats.
+#' * We can't target a named range as the `range`.
+#' * We have no access to the data type of a cell, i.e. we don't know that it's
+#'   logical, numeric, or datetime. That must be re-discovered based on the
+#'   CSV data (or specified by the user).
+#' * Auth and error handling have to be handled a bit differently internally,
+#'   which may lead to behaviour that differs from other functions in
+#'   googlesheets4.
+#'
+#' @section The export URL:
+#'
+#' The export URL supports various query parameters. We always send
+#' `format=csv` and, optionally, `gid=SHEET_ID` and/or `range=A2:D5` (`range`
+#' must really be cell range in this case, not a sheet name or named range):
+#'
+#' ```
+#' # here is the base URL:
+#' https://docs.google.com/spreadsheets/d/SPREADSHEET_ID/export
+#'
+#' # examples of the query part
+#' ?format=csv
+#' ?format=csv?range=CELL_RANGE
+#' ?format=csv?gid=SHEET_ID
+#' ?format=csv?range=A1_range?gid=SHEET_ID
+#' ```
+#'
+#' The export URL supports the use of a bearer token in the Authorization
+#' header.
```

Comment:
The export URL has the nice property that is supports a proper bearer token. My understanding is that this is regarded as a better practice than sending tokens as a query parameter.

Reply:
Oh because this endpoint also works for public sheets?

Reply:
Yeah, I would say both of these URLs are really intended for use with world-readable Sheets.

It's quite hard to find much documentation on the export URL, but I saw evidence somewhere re: bearer token, tried it, and it works.

The docs for the datasource URL explicitly urge people to use it with world-readable Sheets (but the fact that you can send a token via the query is documented).